contracts/RFQ.sol

```diff
@@ -141,18 +195,28 @@ contract RFQ is IRFQ, StrategyBase, ReentrancyGuard, SignatureValidator, BaseLib
             settleAmount = settleAmount.sub(fee);
         }
 
-        // Transfer token/Eth to receiver
+        require(
+            // Confirm that 'makerAddr' sends 'makerAssetAmount' amount of 'makerAssetAddr' to 'address(this)'
+            _order.makerAddr == _makerAssetPermit.user &&
+                _order.makerAssetAddr == _makerAssetPermit.tokenAddr &&
+                address(this) == _makerAssetPermit.recipient &&
+                _order.makerAssetAmount == _makerAssetPermit.amount,
+            "RFQ: maker spender information is incorrect"
+        );
+        // Transfer maker asset to RFQ (= this) first,
```

Comment:
nit: Replace `RFQ`/`this` with `this contract`

`// Transfer maker asset to this contract first`